src/utils/legacyLayerParamUtils.js

```diff
@@ -15,8 +14,12 @@ function readUrlParamValue(url, paramName) {
     return undefined
 }
 
-const newLayerParamRegex = /^[\w.]+[@\w=]*[,ft]*[,?\d.]*$/
-
+/*
+ Return true if the layer is a layer WITH a specified non legacy parameter.
+ Layers without parameters will return false
+ */
+const newLayerParamRegex =
+    /^([\w.-]+)((@[\w=\d]+)+(,[ft,]+)*([,\d.]+)*|(@[\w=\d]+)*(,[ft,]+)+([,\d.]+)*|(@[\w=\d]+)*(,[ft,]+)*([,\d.]+)+)$/
```

Comment:
I don't get the idea of this regex here and the `isLayersUrlParamLegacy()` check below ? @pakb the `handleLegacyParam()` method is called if a legacy url has been detected by a simple check with the missing `#` test, so there is no need for further complex detection on the layer side ? 
However in the legacy url detection I see an issue in the regex, see https://github.com/geoadmin/web-mapviewer/blob/e5ed2ca7ea79f188f875b8a12e91467d8508d5b2/src/router/legacyPermalinkManagement.routerPlugin.js#L21 where we do a global search, this global search regex could match even if it is not containing a `?` at the beginning.
Here I would suggest such regex: `/^\?([^=]+)=([^&]+)/g` but to be honest I don't see the advantage of regex there also could be much simpler as follow:

```javascript
const parseLegacyParams = (search) => {
   if (search.startsWith('?')) {
       return search.split('&').map((p) => p.split('=')).reduce((params, [k, v]) => params[k] = v, {})
   }
   return {}
}
```

Of course this method should be tested with unit test and it would not work if the query parameters would contain any non encoded `&` or `=` in any parameter value, but that should not be the case if the old url are properly encoded.

Reply:
Well, I think you're right : once we're in the legacy handler, we should consider that we are within the supposed use case: someone used an old mapviewer link. I would make the new regex and unit tests associated its own ticket, and I can tackle that once this current issue is resolved.